spec/std/weak_ref_spec.cr

```diff
@@ -0,0 +1,70 @@
+require "spec"
+
+class State
```

Comment:
I would prefer a spec-specific solution, of somehow being able to define helper classes and functions in the namespace of a `describe Foo` statement. This seems possible with macros, and I much prefer it as a solution than having file-private classes and methods.